python/tvm/contrib/torch/as_torch.py

```diff
@@ -0,0 +1,156 @@
+# pylint: disable=missing-function-docstring
+"""
+as_torch: a decorator, which is used to wrap the TVMscript code to `torch.nn.module`.
+"""
+import tempfile
+from typing import Callable, List, Union
+
+import torch
+import torch.utils.dlpack
+
+import tvm
+from tvm.meta_schedule import default_config
+from tvm.meta_schedule.database.database import TuningRecord
+from tvm.meta_schedule.extracted_task import ExtractedTask
+from tvm.meta_schedule.tune import TuneConfig, tune_extracted_tasks
+from tvm.target.target import Target
+from tvm.tir.schedule.schedule import Schedule
+
+
+# python wrapper for OperatorModule
+class OperatorModuleWrapper(torch.nn.Module):
+    def __init__(
+        self,
+        module: Union[
+            tvm.ir.module.IRModule,
+            tvm.tir.function.PrimFunc,
+        ],
+    ):
+        super().__init__()
+        self.rt_module = None  # runtime module
+        self.ir_module = module  # IR modules
+
+    def tune_tir_auto(self, mod: Union[tvm.ir.module.IRModule, tvm.tir.function.PrimFunc]):
+        with tempfile.TemporaryDirectory() as work_dir:
+            sch: Schedule = self.tune_tir_inner(
+                mod=mod,
+                target=Target("llvm --num-cores=16"),
+                work_dir=work_dir,
+            )
+            return sch.mod
+
+    def tune_tir_inner(
+        self,
+        mod: Union[tvm.ir.module.IRModule, tvm.tir.function.PrimFunc],
+        target: Union[str, Target],
+        work_dir: str,
+    ):
+        """Tune a TIR IRModule with a given target.
+
+        Parameters
+        ----------
+        mod : Union[IRModule, PrimFunc]
+            The module to tune.
+        target : Union[str, Target]
+            The target to tune for.
+        work_dir : Optional[str]
+            The working directory to save intermediate results.
+
+        Returns
+        -------
+        sch : Optional[Schedule]
+            The tuned schedule.
+        """
+        mod = default_config.mod(mod)
+        target = default_config.target(target)
+
+        extracted_task = ExtractedTask(
+            task_name="main",
+            mod=mod,
+            dispatched=[mod],
+            target=target,
+            weight=1,
+        )
+        config = TuneConfig(
+            # Default setting
+            strategy="replay_trace",
+            num_trials_per_iter=32,
+            max_trials_per_task=32,
+            max_trials_global=32,
+        )
+        database = tune_extracted_tasks(
+            extracted_tasks=[extracted_task], config=config, work_dir=work_dir
+        )
+        bests: List[TuningRecord] = database.get_top_k(database.commit_workload(mod), top_k=1)
+        if not bests:
+            return None
+        assert len(bests) == 1
+        sch = Schedule(mod)
+        bests[0].trace.apply_to_schedule(sch, remove_postproc=False)
+
+        return sch
+
+    def build(self, target=None):
+        tuned_module = self.tune_tir_auto(self.ir_module)
```

Comment:
If the input TVMScript module doesn't have any tunable knobs, does this `tune_tir_auto` finish instantly? Tuning should be an opt-in feature.